src/components/__tests__/ProseMirror.test.tsx

```diff
@@ -139,6 +140,63 @@ describe("ProseMirror", () => {
     expect(screen.getAllByTestId("paragraph").length).toBeGreaterThanOrEqual(1);
   });
 
+  it("supports composition events", async () => {
+    const schema = new Schema({
+      nodes: {
+        text: {},
+        paragraph: { content: "text*" },
+        doc: { content: "paragraph+" },
+      },
+    });
+    const editorState = EditorState.create({ schema });
+
+    function Paragraph({ children }: NodeViewComponentProps) {
+      return <p data-testid="paragraph">{children}</p>;
+    }
+
+    const reactNodeViews = {
+      paragraph: () => ({
+        component: Paragraph,
+        dom: document.createElement("div"),
+        contentDOM: document.createElement("span"),
+      }),
+    };
+
+    function TestEditor() {
+      const { nodeViews, renderNodeViews } = useNodeViews(reactNodeViews);
+      const [mount, setMount] = useState<HTMLDivElement | null>(null);
+
+      return (
+        <ProseMirror
+          mount={mount}
+          state={editorState}
+          dispatchTransaction={function (this: EditorView, tr) {
+            // We have to wrap the update in `act` to handle all of
+            // the async portal registering and component rendering that
+            // happens "out of band" because it's triggered by ProseMirror,
+            // not React.
+            act(() => this.updateState(this.state.apply(tr)));
```

Comment:
I would think we would want to wrap the `simulateType` call in act, instead, no?

Reply:
I don't think so; simulateType doesn't synchronously trigger the React state update. It changes the DOM, which ProseMirror observes, and then runs dispatchTransaction, which finally updates the state